File: software/matlab/instrumentos/SR830.m
%% LOCKIN SR830
disp('LOCKIN SR830')
disp('MANUAL: http://www.thinksrs.com/downloads/PDFs/Manuals/SR830m.pdf')

%muestro la lista de equipos disponibles, y el string de inicializacion
out=busco_equipos

%inicializo el lockin, conectado con la interfase USB - GPIB
li = gpib('ni',0,8);

%abre la sesión con lockin
fopen(li);

%Leo x: X=1
str=query(li,'outp ?1');
X=str2double(str);

%Leo Y: Y=2
str=query(li,'outp ?2');
Y=str2double(str);

%Leo el modulo: R=3
str=query(li,'outp ?3');
R=str2double(str);

%Leo el ángulo: tita=4
str=query(li,'outp ?4');
tita=str2double(str);

str=query(li, 'SNAP ? 1,2,3,4');%pide las cuatro cosas %X=1, Y=2 ; R=3 , tita=4
data=str2num(str);

%cierra la sesión con lockin
fclose(li);    

File: software/python/instrumentos/SR830.py
# -*- coding: utf-8 -*-
"""
LOCKIN Tektronix SR830
Manual (web): http://www.thinksrs.com/downloads/PDFs/Manuals/SR830m.pdf
Manual (local): \\Srvlabos\manuales\Standford\SR830m.pdf
"""

from __future__ import division, unicode_literals, print_function, absolute_import

import visa

print(__doc__)

# Este string determina el intrumento que van a usar.
# Lo tienen que cambiar de acuerdo a lo que tengan conectado.
resource_name = 'GPIB0::8::INSTR'


rm = visa.ResourceManager()

lockin = rm.open_resource(resource_name)

# Pide indentificacion
print(lockin.query('*IDN?'))

# Lee las salidas una a la vez
# X=1, Y=2, R=3, T=4
x = lockin.query_ascii_values('OUTP ?1')
y = lockin.query_ascii_values('OUTP ?2')
r = lockin.query_ascii_values('OUTP ?3')
t = lockin.query_ascii_values('OUTP ?4')

print(x, y, r, t)

# O bien todas juntas
xyrt = lockin.query_ascii_values('SNAP ? 1,2,3,4')

print(xyrt)

# Cambia el voltaje en la salida auxiliar
# El primer numero es la salida y el segundo es el voltaje
lockin.write('AUXV 0, 4.32')


lockin.close()
